coderd/httpapi/httpapi.go

```diff
@@ -31,16 +29,19 @@ func init() {
 		}
 		return name
 	})
-	err := validate.RegisterValidation("username", func(fl validator.FieldLevel) bool {
+	nameValidator := func(fl validator.FieldLevel) bool {
 		f := fl.Field().Interface()
 		str, ok := f.(string)
 		if !ok {
 			return false
 		}
 		return UsernameValid(str)
```

Comment:
The semantics for username validation may be different to those of workspace and template names.
We should register different validators for each of these.

Reply:
Previously these fields used the `username` validation tag, so behavior hasn't changed and I wasn't looking to change the behavior in this PR. If the validation is incorrect then we should definitely fix that.

Reply:
Related #3321